examples/plot_glm_retinal_ganglia.py

```diff
@@ -0,0 +1,264 @@
+
+The original MATLAB and Python tutorial can be found from
+https://github.com/pillowlab/GLMspiketraintutorial.
+
+These data were collected by Valerie Uzzell in the lab of
+E.J. Chichilnisky at the Salk Institute. For full information see
+Uzzell et al. [1]_, or Pillow et al. [2]_.
+
+In this tutorial, we will demonstrate how to fit linear GLM and
+Poisson GLM to predict the spike counts
+recorded from primate retinal ganglion cells.
+The dataset contains spike responses from 2 ON and 2 OFF parasol
+retinal ganglion cells (RGCs) in primate retina, stimulated with
+full-field `binary white noise`. Two experiment performed consisted of a
+long (20-minute) binary stochastic (non-repeating) stimulus
+which can be used for computing the spike-triggered average
+(or characterizing some other model of the response).
+
+References
+----------
+.. [1] Uzzell, V. J., and E. J. Chichilnisky. "Precision of
+   spike trains in primate retinal ganglion cells."
+   Journal of Neurophysiology 92.2 (2004)
+.. [2] Pillow, Jonathan W., et al. "Prediction and decoding of
+   retinal ganglion cell responses with a probabilistic
+   spiking model." Journal of Neuroscience 25.47 (2005)
+
+"""
+
+# Authors: Jonathan Pillow <pillow@princeton.edu>
+#          Titipat Achakulvisut <my.titipat@gmail.com>
+# License: MIT
+
+########################################################
+#
+# Import all the relevance libraries.
+
+import os.path as op
+import json
+
+import numpy as np
+from scipy.linalg import hankel
+
+from pyglmnet import GLM
+
+import matplotlib.pyplot as plt
+from tempfile import TemporaryDirectory
+
+########################################################
+#
+# Fetch the dataset. The JSON file contains the
+# following keys:  ``stim`` (binary stochastic stimulation),
+# ``stim_times`` (time of the stimulation), and
+# ``spike_times`` (recorded time of the spikes)
+
+from pyglmnet.datasets import fetch_rgc_data
+
+with TemporaryDirectory(prefix="tmp_glm-tools") as temp_dir:
+    dpath = fetch_rgc_data(dpath=temp_dir, accept_rgcs_license=True)
+    with open(op.join(dpath, 'data_RGCs.json'), 'r') as f:
+        rgcs_dataset = json.loads(f.read())
+
+stim = np.array(rgcs_dataset['stim'])
+stim_times = np.array(rgcs_dataset['stim_times'])
+
+n_cells = len(rgcs_dataset['spike_times'])
+dt = stim_times[1] - stim_times[0]  # time between the stimulation
+n_times = len(stim)  # total number of the stimulation
+sfreq = 1. / dt  # frequency of the stimulation
+
+########################################################
+#
+# You can pick a cell to work with and visualize the spikes for one second.
+# In this case, we will pick cell number 2 (ON cell).
+
+cell_idx = 2  # pick cell number 2 (ON cell)
+spike_time = rgcs_dataset['spike_times']['cell_%d' % cell_idx]
+spike_time = np.array(spike_time)
+n_spikes = len(spike_time)  # number of spikes
+
+# bin the spikes
+t_bins = np.arange(n_times + 1) * dt
+spikes_binned, _ = np.histogram(spike_time, t_bins)
```

Comment:
what do you think of calling this variable `y` or at least using `y` when you do `fit` and `predict` so we stay as close to machine learning terminology as possible

Reply:
made changes as you suggested.